zookeeper/task.py

```diff
@@ -1,31 +1,26 @@
 from abc import ABC, abstractmethod
-from zookeeper import Component
-from inspect import getfullargspec, ismethod
+from inspect import signature
+
+from zookeeper.component import Component
 
 
 class Task(Component, ABC):
     """
     A 'Task' component that performs a task on `run`.
     """
 
-    def __init_subclass__(cls: type, *args, **kwargs):
+    def __init_subclass__(cls, *args, **kwargs):
+        super().__init_subclass__(*args, **kwargs)
         # Enforce argument-less `run`
         if hasattr(cls, "run"):
-            call_args = getfullargspec(cls.run).args
-            is_cls_method = ismethod(cls.run) and cls.run.__self__ == cls
-
-            if (
-                len(call_args) == 0
-                or (is_cls_method and call_args == ["cls"])
-                or call_args == ["self"]
-            ):
```

Comment:
The explicit check for class methods appears to be redundant as the explicit tests for that pass. I think it's because the `@classmethod` decorator wraps the function with a normal method which takes a `self` argument? And so `len(call_args) == 1 and "self" in call_args` is true.